lib/compat/wordpress-6.9/command-palette.php

```diff
@@ -60,7 +60,7 @@ function gutenberg_enqueue_command_palette_assets() {
 					if ( preg_match( '/\.php($|\?)/', $submenu_slug ) || wp_http_validate_url( $submenu_slug ) ) {
 						$submenu_url = $submenu_slug;
 					} elseif ( ! empty( menu_page_url( $submenu_slug, false ) ) ) {
-						$submenu_url = menu_page_url( $submenu_slug, false );
+						$submenu_url = html_entity_decode( menu_page_url( $submenu_slug, false ), ENT_QUOTES, get_bloginfo( 'charset' ) );
```

Comment:
I wonder if the logic in `menu_page_url` is the same as https://github.com/WordPress/wordpress-develop/blob/d71b1d89ee166c9685fb357ab02b1260e9b0c8f3/src/wp-admin/menu-header.php#L157-L196

Why isn't it being used in the regular menu? and I wonder if we should have a canonical function about it. 

(No blockers to merge this PR though)

Reply:
The logic used in the `menu-header.php` appears to be much more flexible and robust than the logic in `menu_page_url()`. It would probably be beneficial to extract that logic into a general-purpose function for generating menu URLs, but that's something to consider for future improvements.